src/mysql/result_set.cr

```diff
@@ -76,15 +72,7 @@ class MySql::ResultSet < DB::ResultSet
     if is_nil
       nil
     else
-      yield row_packet, col
+      @columns[col].column_type.read(row_packet)
     end
   end
-
-  {% for t in DB::TYPES %}
-    def read?(t : {{t}}.class) : {{t}}?
-      read_if_not_nil do |row_packet, col|
-        @columns[col].column_type.read(row_packet).as({{t}})
```

Comment:
if each column_type implements a `Column(T)` mixin that stats `def read : T` maybe we could leave

`@columns[col].column_type.as(Column(T)).read(row_packet)`

in order to avoid the implicit unions due to return types.
Yet some overloads for the nillable types will be needed.

@asterite WDYT?

the current PR is not worse that what we currently have due to the implicit unions. So we should be fine.